pkg/gofr/datasource/file/ftp/fs_test.go

```diff
@@ -421,6 +442,15 @@ func TestMkDir(t *testing.T) {
 	}
 }
 
+type directoryAlreadyExistsError struct {
+	code    int
```

Comment:
This error seems to cover more errors than just `directory already exists`.

It should also probably be well marked that this is a test error, and is not expected to be coming from anything other than test code.